sympy/solvers/tests/test_solvers.py

```diff
@@ -683,7 +684,8 @@ def test_issue_5197():
     y = Symbol('y', positive=True)
     # The solution following should not contain {y: -x*exp(x/2)}
     assert solve(x**2 - y**2/exp(x), y, x, dict=True) == [{y: x*exp(x/2)}]
-    assert solve(x**2 - y**2/exp(x), x, y, dict=True) == [{x: 2*LambertW(y/2)}]
```

Comment:
This was not a complete solution. checked from [wolfram](https://www.wolframalpha.com/input/?i=solve+x**2+-+y**2%2Fexp(x)+for+x)

Reply:
But now it returns the complete solution